### PR TITLE
Rename `ResponseCodes` to `ExpectedResponseCodes`

### PR DESCRIPTION
**Pull request by necaris**
Add `ExpectedResponseCodes` and `ExpectedDeleteResponseCodes` inputs, which are more self-explanatory, and mark `ResponseCodes` and `DeleteResponseCodes` as deprecated in field annotations. Also add a `Check()` implementation that ensures only one of the inputs is set, and sets the new field appropriately so we can use the new field consistently elsewhere in the code.

Follow-up to #75, from feedback.

### CODE REVIEW

**Review by dirien — CHANGES_REQUESTED** [review]
Hi @necaris,

can you rebase the PR please? Then I can start the review!

File: provider/cmd/pulumi-resource-purrl/schema.json [hunk 1]
```diff
@@ -97,7 +97,7 @@
                     "type": "string"
                 },
                 "deleteResponseCodes": {
-                    "description": "The expected response code.",
+                    "description": "The expected response code(s) for deletion. Deprecated -- use `expectedDeleteResponseCodes` instead.",
```
Comment (dirien) [inline provider/cmd/pulumi-resource-purrl/schema.json:100]: Hi @necaris,

can you use `deprecationMessage` instead?

see  https://www.pulumi.com/docs/guides/pulumi-packages/schema/#property

File: provider/cmd/pulumi-resource-purrl/schema.json [hunk 2]
```diff
@@ -131,7 +145,7 @@
                     "type": "string"
                 },
                 "responseCodes": {
-                    "description": "The expected response code.",
+                    "description": "The expected response code(s). Deprecated -- use `expectedResponseCodes` instead.",
```
Comment (dirien) [inline provider/cmd/pulumi-resource-purrl/schema.json:148]: Hi @necaris,

can you use `deprecationMessage` instead?

see  https://www.pulumi.com/docs/guides/pulumi-packages/schema/#property

File: provider/cmd/pulumi-resource-purrl/schema.json [hunk 3]
```diff
@@ -187,11 +201,11 @@
                     "type": "string"
                 },
                 "deleteResponse": {
-                    "description": "The response from the API call.",
+                    "description": "The response from the (delete) API call.",
                     "type": "string"
                 },
                 "deleteResponseCodes": {
-                    "description": "The expected response code.",
+                    "description": "The expected response code(s) for deletion. Deprecated -- use `expectedDeleteResponseCodes` instead.",
```
Comment (dirien) [inline provider/cmd/pulumi-resource-purrl/schema.json:208]: Hi @necaris,

can you use `deprecationMessage` instead?

see  https://www.pulumi.com/docs/guides/pulumi-packages/schema/#property

File: provider/cmd/pulumi-resource-purrl/schema.json [hunk 4]
```diff
@@ -232,7 +260,7 @@
                     "type": "integer"
                 },
                 "responseCodes": {
-                    "description": "The expected response code.",
+                    "description": "The expected response code(s). Deprecated -- use `expectedResponseCodes` instead.",
```
Comment (dirien) [inline provider/cmd/pulumi-resource-purrl/schema.json:263]: Hi @necaris,

can you use `deprecationMessage` instead?

see  https://www.pulumi.com/docs/guides/pulumi-packages/schema/#property

**Review by dirien — CHANGES_REQUESTED** [review]
hey @necaris,

thanks for rebasing this. I left a little review and after this we are ready to merge this!

**Review by dirien — COMMENTED** [review]
Hi @necaris,

thanks for the rebase. I tested the new code with following example:

```go
func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		purrlCommand, err := purrl.NewPurrl(ctx, "httpbin", &purrl.PurrlArgs{
			Url:  pulumi.String("https://httpbin.org/get"),
			Name: pulumi.String("httpbin"),
			ExpectedResponseCodes: pulumi.StringArray{
				pulumi.String("200"),
			},
			Method: pulumi.String("GET"),
			Headers: pulumi.StringMap{
				"test": pulumi.String("test"),
			},
			DeleteMethod: pulumi.String("DELETE"),
			DeleteUrl:    pulumi.String("https://httpbin.org/delete"),
			ExpectedDeleteResponseCodes: pulumi.StringArray{
				pulumi.String("200"),
			},
		})
		if err != nil {
			return err
		}
		ctx.Export("actual response code", purrlCommand.ResponseCode)
		return nil
	})
}
```

And I get following message:

```bash
View in Browser (Ctrl+O): https://app.pulumi.com/dirien/purrl-go/dev/previews/475953b4-4093-49b8-9b8f-f98cbf73725c

     Type                  Name          Plan     Info
     pulumi:pulumi:Stack   purrl-go-dev           2 messages
     └─ purrl:index:Purrl  httpbin                1 error


Diagnostics:
  purrl:index:Purrl (httpbin):
    error: at least one of deleteResponseCodes or expectedDeleteResponseCodes must be provided

  pulumi:pulumi:Stack (purrl-go-dev):
```

I think the `Check` function is not proper working.

**Comment by necaris** [comment]
Thanks @dirien for the review! Let me look at it.

**Comment by dirien** [comment]
Hi @necaris,

do you need any additional feedback from me for this?

**Comment by dirien** [comment]
Hey @necaris,

closing this PR for now. Feel free to open a new one if necessary!

Thanks!

**Pull request closed without merging**